### PR TITLE
Eslint: enable 'no-underscore-dangle' to unify naming

### PR DESCRIPTION
**Pull request by mrtnzlml**
**Proposal**: I can see that underscore may be good visual clue providing information
about method being sort of private. But I have two arguments against:

1) Underscore method is not really private.
2) All React lifecycle methods are meant to be private but they are
not prefixed by the underscore which creates inconsistency and
additional confusion.

See: https://reactjs.org/docs/react-component.html#the-component-lifecycle
See: https://eslint.org/docs/rules/no-underscore-dangle
66646d5

### CODE REVIEW

**Comment by mrtnzlml** [comment]
Please test these changes in [Expo](https://docs.expo.io/versions/latest/introduction/installation.html#mobile-client-expo-for-ios-and-android) after build has finished:

  QR Code

File: app/core/src/components/authentication/EmailLoginForm.js [hunk 1]
```diff
@@ -24,18 +24,18 @@ export default class SimpleLoginForm extends React.Component<Props, State> {
...
   };
 
-  _handleEmailChange = text =>
+  handleEmailChange = (text: string) =>
     this.setState({
       username: text,
     });
 
-  _handlePasswordChange = text =>
+  handlePasswordChange = (text: string) =>
```
Comment (mrtnzlml) [inline app/core/src/components/authentication/EmailLoginForm.js]: I didn't know about this. Flow doesn't care about underscored methods. This is not what we want right? 💩 🙂

**Comment by mrtnzlml** [comment]
Also, this helped to discover an error in Flow setting (underscored methods were ignored)... 💩

**Comment by jaroslav-kubicek** [comment]
there are some conflicts, cc @mrtnzlml

**Comment by mrtnzlml** [comment]
> there are some conflicts, cc @mrtnzlml

Resolved.

**Pull request merged**